libraries/chain/include/eosio/chain/global_property_object.hpp

```diff
@@ -45,11 +45,13 @@ namespace eosio { namespace chain {
       optional<block_num_type>            proposed_schedule_block_num;
       shared_producer_authority_schedule  proposed_schedule;
       chain_config                        configuration;
+      chain_id_type                       chain_id;
```

Comment:
Adding this field means you will also need to make changes to the `state_history_plugin`. As it is currently, it will not reflect this new field. The good news is that the new WTMSig block producer authority changes introduced in https://github.com/EOSIO/eos/pull/7404 added the new `global_property_v1` type to SHiP that you need. So since this version hasn't been released yet, you can just piggyback off of that. This will require changing the serialization code for `global_property_object` within `state_history_serialization.hpp` to serialize the `chain_id` field as well, as well as adding the field into the ABI for `global_property_v1` struct in `state_history_plugin_abi.cpp`.

Reply:
I have also asked @tbfleming to review this PR so we can get another set of eyes on this.

Reply:
@arhag is correct. This will be an easy change.